### PR TITLE
Added build log generation/uploading

### PR DESCRIPTION
**Pull request by jaydee029**
For issues
- galasa-dev/projectmanagement#1965
- galasa-dev/projectmanagement#1967

### CODE REVIEW

**Comment by galasa-team** [comment]
Automatic triggering of the build is cancelled as user is not a member of the approved groups. Please ask an admin to review this PR and create a comment on the PR stating 'Approved for building'

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -31,25 +31,41 @@ jobs:
...
                 -PcentralMaven=https://repo.maven.apache.org/maven2/ \
-                -PtargetMaven=${{github.workspace}}/repo
+                -PtargetMaven=${{github.workspace}}/repo 2>&1 | tee gradle-build.log
+              
+            - name: Upload Maven Build Log
+              if: always()
+              uses: actions/upload-artifact@v4
+              with:
+                name: gradle-build-log
+                path: gradle-build.log
```
Comment (jadecarino) [inline .github/workflows/build.yaml:49]: I think this is meant to say Gradle here. Also can we rename the name of the artefact to "inttests-gradle-build-log" here so we can differentiate from what is being built below...

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -31,25 +31,41 @@ jobs:
...
                   --batch-mode --errors --fail-at-end \
-                  --settings ${{github.workspace}}/settings.xml
+                  --settings ${{github.workspace}}/settings.xml 2>&1 | maven-build.log
+            
+            - name: Upload Maven Build Log
+              if: always()
+              uses: actions/upload-artifact@v4
+              with:
+                name: maven-build-log
+                path: maven-build.log
```
Comment (jadecarino) [inline .github/workflows/build.yaml:68]: Can we rename this to "inttests-obr-maven-build-log" so we can differentiate from the above. The first build is to build the source code of the inttests and this is to build the inttests OBR.

**Review by jadecarino — CHANGES_REQUESTED** [review]
A couple of minor changes for clarity and as with the other PRs, please can we change the upload artefacts to only happen on failure, and add the same into the PR builds, both with a retention period of 7 days

File: .github/workflows/build.yaml (hunk 1 above)
Comment (jaydee029) [reply .github/workflows/build.yaml]: done

File: .github/workflows/build.yaml (hunk 2 above)
Comment (jaydee029) [reply .github/workflows/build.yaml]: done

**Comment by jaydee029** [comment]
> A couple of minor changes for clarity and as with the other PRs, please can we change the upload artefacts to only happen on failure, and add the same into the PR builds, both with a retention period of 7 days

done

**Comment by galasa-team** [comment]
Automatic triggering of the build is cancelled as user is not a member of the approved groups. Please ask an admin to review this PR and create a comment on the PR stating 'Approved for building'

**Comment by galasa-team** [comment]
Automatic triggering of the build is cancelled as user is not a member of the approved groups. Please ask an admin to review this PR and create a comment on the PR stating 'Approved for building'

**Review by jadecarino — APPROVED** [review]
Approved for building

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**